Unit test flake: TestCheckVersionCompatibility() Expected value not to be nil

The recently merged PR #21741 added a version cache in dockertools, which makes the test [`TestCheckVersionCompatibility`](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/dockertools/manager_test.go#L1856-L1895) flaky.
```
--- FAIL: TestCheckVersionCompatibility (0.01s)
         <autogenerated>:31:                                 Error Trace:    manager_test.go:1892
                 Error:          Expected value not to be nil.
                 Messages:       test case #0 test version "1.20" version error check

FAIL
```

The reason is that `TestCheckVersionCompatibility` injected error into fake_docker_client and then expected to see the error from `fake_docker_client.Version()` right after that (See [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/dockertools/manager_test.go#L1890-L1892)). However, the error was consumed by the goroutine which periodically updates the version cache (See [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/dockertools/manager.go#L262)).

The docker daemon version is rarely updated, and we only need version once for each container creating, during which we also do container create, start, inspect  #etc.
1) Do we care about the cost of the `Version` call so much that we have to introduce the complexity?
2) Kubelet periodically calls `runtime.Status()`, can we rely on that to update the cache? 

Anyhow, at least, we should disable the cache or have a fake one in the unit test. :)

@yujuhong @vishh @resouer 
/cc @kubernetes/sig-node

--- a/pkg/kubelet/dockertools/fake_manager.go
+++ b/pkg/kubelet/dockertools/fake_manager.go
@@ -24,7 +24,6 @@ import (
 	"k8s.io/kubernetes/pkg/kubelet/network"
 	proberesults "k8s.io/kubernetes/pkg/kubelet/prober/results"
 	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
-	"k8s.io/kubernetes/pkg/kubelet/util/cache"
 	"k8s.io/kubernetes/pkg/types"
 	"k8s.io/kubernetes/pkg/util/flowcontrol"
 	"k8s.io/kubernetes/pkg/util/oom"
@@ -53,9 +52,6 @@ func NewFakeDockerManager(
 		burst, containerLogsDir, osInterface, networkPlugin, runtimeHelper, httpClient, &NativeExecHandler{},
 		fakeOOMAdjuster, fakeProcFs, false, imageBackOff, false, false, true)
 	dm.dockerPuller = &FakeDockerPuller{}
-	dm.versionCache = cache.NewVersionCache(func() (kubecontainer.Version, kubecontainer.Version, error) {
-		return dm.getVersionInfo()
-	})
 	return dm
 }
 

--- a/pkg/kubelet/dockertools/manager.go
+++ b/pkg/kubelet/dockertools/manager.go
@@ -253,15 +253,6 @@ func NewDockerManager(
 		optf(dm)
 	}
 
-	// initialize versionCache with a updater
-	dm.versionCache = cache.NewVersionCache(func() (kubecontainer.Version, kubecontainer.Version, error) {
-		return dm.getVersionInfo()
-	})
-	// update version cache periodically.
-	if dm.machineInfo != nil {
-		dm.versionCache.UpdateCachePeriodly(dm.machineInfo.MachineID)
-	}
-
 	return dm
 }
 
@@ -1581,26 +1572,13 @@ func (dm *DockerManager) calculateOomScoreAdj(container *api.Container) int {
 	return oomScoreAdj
 }
 
-// getCachedVersionInfo gets cached version info of docker runtime.
-func (dm *DockerManager) getCachedVersionInfo() (kubecontainer.Version, kubecontainer.Version, error) {
-	apiVersion, daemonVersion, err := dm.versionCache.Get(dm.machineInfo.MachineID)
-	if err != nil {
-		glog.Errorf("Failed to get cached docker api version %v ", err)
-	}
-	// If we got nil versions, try to update version info.
-	if apiVersion == nil || daemonVersion == nil {
-		dm.versionCache.Update(dm.machineInfo.MachineID)
-	}
-	return apiVersion, daemonVersion, err
-}
-
 // checkDockerAPIVersion checks current docker API version against expected version.
 // Return:
 // 1 : newer than expected version
 // -1: older than expected version
 // 0 : same version
 func (dm *DockerManager) checkDockerAPIVersion(expectedVersion string) (int, error) {
-	apiVersion, _, err := dm.getCachedVersionInfo()
+	apiVersion, _, err := dm.getVersionInfo()
 	if err != nil {
 		return 0, err
 	}

